mspray/apps/main/tests/views/test_performance.py

```diff
@@ -1,36 +1,290 @@
+        report2.found = 12
+        report2.reported_sprayed = 6
+        report1.district = district
+        report2.save()
+
         factory = RequestFactory()
         request = factory.get("/")
         view = DistrictPerfomanceView.as_view()
         response = view(request)
         self.assertEqual(response.status_code, 200)
 
+        # Query obtains all data for the SprayOperator
+        # including submissions made and passes this to the serializer
+
+        queryset = Location.objects.raw(DISTRICT_PERFORMANCE_SQL)
+        serializer = DistrictPerformanceReportSerializer(queryset, many=True)
+
+        self.assertEqual(response.context_data['data'], serializer.data)
+
     @override_settings(MSPRAY_SUPERVISOR_LABEL='Supervisor')
     def test_team_leader_performance(self):
         """Test team leaders performance view."""
         data_setup()
-        sprayarea = Location.objects.get(name='Akros_1')
+        self._load_fixtures()
+        rhc = Location.objects.get(name='Miti')
+        district = rhc.parent
+        spray_operator = SprayOperator.objects.first()
+        team_leader = TeamLeader.objects.first()
+        spray_operator.team_leader = team_leader
+        spray_operator.rhc = rhc
+        spray_operator.save()
+
+        # next we identify a SprayDay object for the spray operator
+        spray_day = SprayDay.objects.filter(spray_operator=spray_operator)
+        spray_day.update(sprayable=True)
+
+        # Create performance report objects from the submissions made
+        # by the particular sprayoperator.
+
+        performance_report(spray_operator)
+        report1 = PerformanceReport.objects.get(spray_operator=spray_operator)
+        report1.found = 7
+        report1.reported_sprayed = 16
+        report1.reported_found = 22
+        report1.district = rhc
+        report1.save()
+
+        report2 = PerformanceReport.objects.get(spray_operator=spray_operator)
+        report2.id = None
+        report2.sprayformid = 7658
+        report2.refused = 6
+        report2.found = 12
+        report2.reported_sprayed = 6
+        report1.district = rhc
+        report2.save()
 
         factory = RequestFactory()
         request = factory.get("/performance/team-leaders/458")
         view = TeamLeadersPerformanceView.as_view()
-        response = view(request, slug=sprayarea.id)
+        response = view(request, slug=district.id)
         self.assertEqual(response.status_code, 200)
         self.assertTrue('Supervisor' in str(response.render().content))
+
+        queryset = TeamLeaderAssistant.objects.raw(
+            TLA_PERFORMANCE_SQL, [district.id]
+        )
+        serializer = TLAPerformanceReportSerializer(queryset, many=True)
+
+        result = {
+            'other': 0,
+            'refused': 6,
+            'sprayed': 2,
+            'sprayable': 19,
+            'not_sprayable': 0,
+            'not_eligible': 0,
+            'not_sprayed_total': 6,
+            'data_quality_check': False,
```

Comment:
Why is the data quality check False in this case?

Reply:
@moshthepitt the data quality check is not a computated value from the serializer, rather it is picked from the last performance report object submitted by the spray_operator(distinguished by the spray_date).
Thus, since when creating the test case performance report objects I did not specify the boolean value to the 'True'...it was set to by default to be false, and eventually the output from the serializer was False.